openassessment/xblock/ui_mixins/legacy/handlers_mixin.py

```diff
@@ -61,6 +63,26 @@ class LegacyHandlersMixin:
     Exposes actions (@XBlock.json_handlers) used in our legacy ORA UI
     """
 
+    @staticmethod
+    def send_ora_submission_created_event(submission: dict) -> None:
+        """
+        Send an event when a submission is created
+
+        Args:
+            submission (dict): The submission data
+        """
```

Comment:
This change means we're throwing 2 "events" on ORA submission, this new one plus [the legacy analytics event](https://github.com/eduNEXT/edx-ora2/blob/00f9f65081700db95820210e12eb4308196a8afc/openassessment/xblock/apis/submissions/submissions_actions.py#L161-L171).

I'm a little worried that this will be confusing to people trying to consume these events, since you can't get everything you need from either event (this one has file urls but not the text answer, the analytics event has the text answer and other useful fields, but no file urls).

But I'm not sure how best to resolve this.. Any thoughts @BryanttV @mariajgrimaldi ?

Maybe instead of [having your plugin re-fetch the submission in when it receives this new event](https://github.com/eduNEXT/platform-plugin-turnitin/blob/b513df6fd20c6c36d0a6ac743dce9965127779c9/platform_plugin_turnitin/tasks.py#L37), you could include all the extra submission data in the `ORASubmissionData`? Then we could say in a comment here that `ORA_SUBMISSION_CREATED` should be used instead of the old `openassessmentblock.create_submission` analytics event.

Reply:
That sounds great thank you @mariajgrimaldi and @BryanttV !

Reply:
FYI, I am on leave from 20 - 29 April, so if we need to merge this now to make the redwood cut, it's fine to make this data change part of a separate PR.

If that's the way you'd rather go, please let me know. You'll just need to resolve the conflicts after we merge https://github.com/openedx/edx-ora2/pull/2204, and I can merge this first thing tomorrow (around 00:00 UTC).

Reply:
We'll update this PR with the new data class today. We'll let you know how it goes before 00:00 UTC!

Reply:
@pomegranited now the data sent by the Open edX Event is a superset of what the legacy analytics event sends:

https://github.com/openedx/openedx-events/blob/main/openedx_events/learning/data.py#L450C1-L495C47

So, enough information is available for the events' users, and if, in the foreseeable future, the analytics event is replaced by this one, then the same data will be available. 

Thank you, @BryanttV, for all the hard work! And to you, Jill, for the detailed review. Please, let us know what you think.

Reply:
Your changes look great, thank you! Will just run some quick manual tests and get this merged.